### PR TITLE
Change bundle name rules on iOS to match Apple's requirements

### PR DESCRIPTION
**Pull request by NoFr1ends**
**Fix** #34302 

---
Changed the bundle identifier verification to match the official verficiation.
https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleidentifier

This help developers to migrate existing games to godot without
constantly having to change the bundle identifier in the generated
XCode project.

### CODE REVIEW

File: platform/iphone/export/export.cpp [hunk 1]
```diff
@@ -114,54 +114,14 @@ class EditorExportPlatformIOS : public EditorExportPlatform {
...
-				}
-				return false;
-			}
-			if (c == '.') {
-				segments++;
-				first = true;
-				continue;
-			}
-			if (!((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '-')) {
+			if (!((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '-') || c == '.') {
```
Comment (akien-mga) [inline platform/iphone/export/export.cpp]: Won't this prevent using `.`, and thus reverse DNS identifiers?

**Comment by akien-mga** [comment]
> Changed the bundle identifier verification to match the official verficiation.
https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleidentifier

BTW, that page does state:

> The string should be in reverse-DNS format.

But if they don't enforce it, I guess we shouldn't either.

File: platform/iphone/export/export.cpp (hunk 1 above)
Comment (NoFr1ends) [reply platform/iphone/export/export.cpp]: 🤦‍♂ fixed..

**Comment by NoFr1ends** [comment]
And yes that's what I said in the issue too.

File: platform/iphone/export/export.cpp (hunk 1 above)
Comment (akien-mga) [reply platform/iphone/export/export.cpp]: Hm, it still seems to be the same to me. In pseudo code it's `if (!(is_alphanumeric()) || c == '.') error`.

The `|| c == '.'` should be in the `!(  )`.

Comment (NoFr1ends) [reply platform/iphone/export/export.cpp]: I don't know why but it seemed like git didn't added the changes (or maybe I just forget it), it's correctly now.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**